apm-sniffer/apm-sdk-plugin/logger-plugin/src/main/java/org/apache/skywalking/apm/plugin/logger/Log4j2LoggerInterceptor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.logger;
+
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+
+public class Log4j2LoggerInterceptor implements InstanceMethodsAroundInterceptor {
+
+    private static final ContextConfig.LoggerConfig CONFIG = ContextConfig.getInstance().getLog4j2Config();
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        String loggerName = objInst.getSkyWalkingDynamicField().toString();
```

Comment:
In the real world coding, there is a `log level` config in the app code level, such as in `log4j.xml`. What do you expect if it mismatched w/ plugin's config?
In the current implementation, you are overriding it. Is this a good way? @vcjmhg @BFergerson

Reply:
I think the argument could be made that these log levels have different contexts and therefore must be configured independently. What is the alternative? Waiting till the user's logging framework is fully loaded and using that to generate the monitored log configuration?

Reply:
I think the most confusing thing would be `2 configurations`. Eventually, I would expect the Satellite project to take the responsibility of the logging collecting, rather than using tracing context. 
In the short term, we could try to use this, as it is simple enough. 

And, log level should be available when the interceptor invoking, right? How about we use it directly? such as `this#isDebugEnabled`?
The idea is from the performance perspective. If users think logging to file or logging system had over-load risk, it definitely should have more risks to collect those through tracing core. Right?

So, we could have our config file but basically should have lower-priority comparing to the official log config file(such as `log4j.xml`). Is this better? Of course, if we provide that, we could have a `collecting all loggable things` configuration in default, such as no config file, mean while this plugin must be placed in the optional plugin to avoid OOM.

Reply:
FYI @EvanLjp, we are doing log collection for now.

Reply:
@wu-sheng, I've made the changes you suggested [here](https://github.com/vcjmhg/skywalking/pull/9/files). This makes the monitored logging configuration a lower priority than the official log config.